tests/templates/test_state_preparations/test_qutrit_basis_state_prep.py

```diff
@@ -0,0 +1,217 @@
+        ([0, 0, 1, 0], [1, 2, 3, 4], [3]),
+        ([1, 1, 1, 0], [1, 2, 6, 8], [1, 2, 6]),
+        ([1, 0, 1, 1], [1, 2, 6, 8], [1, 6, 8]),
+    ])
+    # fmt: on
+    def test_correct_pl_gates(self, basis_state, wires, target_wires):
+        """Tests queue for simple cases."""
+
+        op = qml.QutritBasisStatePreparation(basis_state, wires)
+        queue = op.expand().operations
+
+        for id, gate in enumerate(queue):
+            assert gate.name == "TShift"
+            assert gate.wires.tolist() == [target_wires[id]]
+
+    # fmt: off
+    @pytest.mark.parametrize("basis_state,wires,target_state", [
+        ([0], [0], [0, 0, 0]),
+        ([0], [1], [0, 0, 0]),
+        ([1], [0], [1, 0, 0]),
+        ([1], [1], [0, 1, 0]),
+        ([0, 1], [0, 1], [0, 1, 0]),
+        ([1, 1], [0, 2], [1, 0, 1]),
+        ([1, 1], [1, 2], [0, 1, 1]),
+        ([1, 0], [0, 2], [1, 0, 0]),
+        ([1, 1, 0], [0, 1, 2], [1, 1, 0]),
+        ([1, 0, 1], [0, 1, 2], [1, 0, 1]),
+    ])
+    # fmt: on
+    def test_state_preparation(self, tol, qutrit_device_3_wires, basis_state, wires, target_state):
+        """Tests that the template produces the correct expectation values."""
+
+        @qml.qnode(qutrit_device_3_wires)
+        def circuit(obs):
+            qml.QutritBasisStatePreparation(basis_state, wires)
+
+            return [qml.expval(qml.THermitian(A=obs, wires=i)) for i in range(3)]
+
+        # Convert to basis states
+        obs = np.array([[1, 0, 0], [0, 2, 0], [0, 0, 3]])
+        output_state = [x - 1 for x in circuit(obs)]
+
+        assert np.allclose(output_state, target_state, atol=tol, rtol=0)
+
+    # @pytest.mark.jax
+    # @pytest.mark.parametrize(
+    #     "basis_state,wires,target_state",
+    #     [
+    #         ([0, 1], [0, 1], [0, 1, 0]),
+    #         ([1, 1, 0], [0, 1, 2], [1, 1, 0]),
+    #         ([1, 0, 1], [2, 0, 1], [0, 1, 1]),
+    #     ],
+    # )
+    # def test_state_preparation_jax_jit(
+    #     self, tol, qutrit_device_3_wires, basis_state, wires, target_state
+    # ):
+    #     """Tests that the template produces the correct expectation values."""
+    #     import jax
+
+    #     @qml.qnode(qutrit_device_3_wires, interface="jax")
+    #     def circuit(state, obs):
+    #         qml.QutritBasisStatePreparation(state, wires)
+
+    #         return [qml.expval(qml.THermitian(A=obs, wires=i)) for i in range(3)]
+
+    #     circuit = jax.jit(circuit)
+
+    #     obs = np.array([[1, 0, 0], [0, 2, 0], [0, 0, 3]])
+    #     output_state = [x - 1 for x in circuit(basis_state, obs)]
+
+    #     assert np.allclose(output_state, target_state, atol=tol, rtol=0)
+
+    # @pytest.mark.tf
+    # @pytest.mark.parametrize(
+    #     "basis_state,wires,target_state",
+    #     [
+    #         ([0, 1], [0, 1], [0, 1, 0]),
+    #         ([1, 1, 0], [0, 1, 2], [1, 1, 0]),
+    #         ([1, 0, 1], [2, 0, 1], [0, 1, 1]),
+    #     ],
+    # )
+    # def test_state_preparation_tf_autograph(
```

Comment:
This test passes locally for me - was it raising an error for you?

Reply:
I actually left it out because I could not check it out locally since I could not find a working installation of Tensorflow for M1 Mac architecture. I also thought `default_qubit_tf/jax` should be implemented for these two tests to make sense. Is that not the case?

Reply:
Well, I am uncommented now, and it is not passing the test. I honestly don't know what can be going wrong since I cannot test it locally and I don't quite understand from the pipeline test where is the error coming from.

Reply:
You are absolutely right, I ran the qubit test file. The jax and tf interfaces aren't implemented for qutrits, just mark both as xfail!

If you want `tensorflow` on your M1 Mac, I used this install guide for mine: https://caffeinedev.medium.com/how-to-install-tensorflow-on-m1-mac-8e9b91d93706

But I believe you have to be using conda via Miniforge for it to work, I'll leave it up to you whether reconfiguring your setup is worth it just for `tensorflow` 😅

Reply:
Done! And thank you for the TF link, this is gonna be super useful! If you have a similar one for JAX that works with Pennylane, that would also be great :) (between numpy, jax, and jaxlib incompatibilities, I found it impossible to get them all working at the same time)

Reply:
Once I had the conda miniforge stuff set up I just pip-installed those ones, but there can be some issues with versions for sure. I currently have:

tensorflow-macos: 2.12.0
jax and jaxlib: 0.4.10
numpy: 1.23.5

and they seem to all be playing nicely together. I actually changed my numpy version a few days ago because it wasn't getting along with tensorflow 🥲

Reply:
It can really become super messy to try to make them work together 😪.
I will keep the link and versions you sent me somewhere safe for when I next need it 😆. Thank you! :D